### DISCUSSION TITLE
How to obtain the raw linear values of every swatches?

### DISCUSSION THREAD

**Discussion by Hurricane-k** [colour-science/colour-checker-detection#84] [Q&A]
### Question

Hi

I following the `.ipynb` about segmentation. When I using
```
detect_colour_checkers_segmentation(
        COLOUR_CHECKER_IMAGES, additional_data=True)
```
I hope I can get the absolute locations of every patch. I think `quadrilateral ` of `Tuple[DataDetectionColourChecker]` can help me. 
However, even if parameter `additional_data` is set as `True`, I cannot get the last element 'quadrilateral'. I only get the first three elements `swatch_colors`, 'swatch_masks', 'colour_checker'.

How can I get the `quadrilateral`, or how can I get the absolute locations of every color patch?

Best Regards.

**Comment by Hurricane-k** [comment]
Actually, I wanna get the linear raw rgb values of every swatch. And my input is raw `.png` file without WB, and I use other default parameters in `detect_colour_checkers_segmentation`. But I check the results `swatch_colours` that are far away from my manual operation. I think there might be some secrets or tricks I missed. Surprisingly, I found this discussion [#59](https://github.com/colour-science/colour-checker-detection/discussions/59). the input image must be sRGB or other standard device-independence color space. coz my aim is to get the exact raw value. how do I do?

**Comment by Hurricane-k** [comment]
Another question: 
At the every beginning, I completely follow the instruction of segmentation (jupyter notebook example), which means I use `colour.cctf_decoding(colour.io.read_image(path))` to load IMG. 
Based on the discussion [#59](https://github.com/colour-science/colour-checker-detection/discussions/59),  About `colour.cctf_decoding`, its description is decoding non-linear RGB values to linear RGB, could you explain further the exact definiton of this decoding operion, is it inverse operation of Gamma correction?

Other interesting thing: when I load linear RAW `.png` directly, the recognition is not really good. 
![image](https://github.com/user-attachments/assets/5176ad7d-05ad-4c06-8c0b-fd566eb8901b)

But when I using `colour.cctf_decoding(colour.io.read_image(path))` , it manage to detect color checker.
![image](https://github.com/user-attachments/assets/80d0acfe-f3c1-41b4-be1c-c5946bce8f0d)

**Comment by KelSolaar** [comment]
Hello,

Typically, a PNG file and because it tends to be 8-bit will be saved with a non-linear encoding to avoid quantization issues. An EXR 16-bit Half or 32-bit file would be linearly encoded on the other hand. You can call this *gamma correction* but usually the encodings are more complex than a simple power function.

How did you obtain/produce your RAW png file?

**Comment by Hurricane-k** [comment]
Hi

Thank you for your quick response. 

From `*.CR2` or `*.DNG` file, and then convert them to de-mosaicked images by `dcraw`.

**Comment by Hurricane-k** [comment]
Hi

in your example, `examples_detection_segmentation.ipynb`, 

```
for image in COLOUR_CHECKER_IMAGES:
      for colour_checker_data in detect_colour_checkers_segmentation(
            image, additional_data=True):

            swatch_colours, swatch_masks, colour_checker_image = (
                   colour_checker_data.values)
```

why only three variables `swatch_colours, swatch_masks, colour_checker_image`?

Based on [Reference API](https://colour-checker-detection.readthedocs.io/en/latest/generated/colour_checker_detection.detect_colour_checkers_segmentation.html), return Tuple[[colour_checker_detection.detection.DataDetectionColourChecker](https://colour-checker-detection.readthedocs.io/en/latest/generated/colour_checker_detection.detection.DataDetectionColourChecker.html#colour_checker_detection.detection.DataDetectionColourChecker)], there are four elements in the tuple, how to get the fourth variant `quadrilateral`?

Best Regards

**Comment by Hurricane-k** [comment]
> Hi
> 
> in your example, `examples_detection_segmentation.ipynb`,
> 
> ```
> for image in COLOUR_CHECKER_IMAGES:
>       for colour_checker_data in detect_colour_checkers_segmentation(
>             image, additional_data=True):
> 
>             swatch_colours, swatch_masks, colour_checker_image = (
>                    colour_checker_data.values)
> ```
> 
> why only three variables `swatch_colours, swatch_masks, colour_checker_image`?
> 
> Based on [Reference API](https://colour-checker-detection.readthedocs.io/en/latest/generated/colour_checker_detection.detect_colour_checkers_segmentation.html), return Tuple[[colour_checker_detection.detection.DataDetectionColourChecker](https://colour-checker-detection.readthedocs.io/en/latest/generated/colour_checker_detection.detection.DataDetectionColourChecker.html#colour_checker_detection.detection.DataDetectionColourChecker)], there are four elements in the tuple, how to get the fourth variant `quadrilateral`?
> 
> Best Regards

no `Tuple` of `DataDectectionColourChecker` when passing `additional_data=True`

**Comment by KelSolaar** [comment]
Hello,

Sorry, I was abroad in US, which version of the package are you using? From `develop` or `pypi`?

**Reply by Hurricane-k** [reply]
Hi,

Thank you for your time and reply. From `pypi`
```
===============================================================================
*                                                                             *
*   Interpreter :                                                             *
*       python : 3.11.6 | packaged by conda-forge | (main, Oct  3 2023,       *
*   10:40:35) [GCC 12.3.0]                                                    *
*                                                                             *
*   colour-science.org :                                                      *
*       colour : 0.4.4                                                        *
*       colour-checker-detection : 0.2.0                                      *
*                                                                             *
*   Runtime :                                                                 *
*       imageio : 2.35.1                                                      *
*       matplotlib : 3.8.2                                                    *
*       networkx : 3.2.1                                                      *
*       numpy : 1.26.4                                                        *
*       pandas : 2.1.4                                                        *
*       scipy : 1.14.1                                                        *
*       tqdm : 4.66.1                                                         *
*       opencv : 4.10.0                                                       *
*                                                                             *
===============================================================================
```

**Reply by KelSolaar** [reply]
Ah yeah that makes sense, so the docs you are referring to are for the `develop` branch on Github.

**Reply by Hurricane-k** [reply]
OK, I'll try `dev`. Thank you.
